tests/lib/rules/sort-comp.js

```diff
@@ -91,6 +91,31 @@ ruleTester.run('sort-comp', rule, {
         'everything-else'
       ]
     }]
+  }, {
+    // Must validate a full React 16.3 class
+    code: [
+      'class Hello extends React.Component {',
+      '  static displayName = \'\'',
+      '  static propTypes = {}',
+      '  static defaultProps = () => {}',
+      '  constructor() {}',
+      '  myInstanceVariable = \'\'',
+      '  state = {}',
+      '  static getDerivedStateFromProps = () => {}',
+      '  UNSAFE_componentWillMount = () => {}',
+      '  componentDidMount = () => {}',
+      '  UNSAFE_componentWillReceiveProps = () => {}',
+      '  shouldComponentUpdate = () => {}',
+      '  UNSAFE_componentWillUpdate = () => {}',
+      '  getSnapshotBeforeUpdate = () => {}',
+      '  componentDidUpdate = () => {}',
+      '  componentWillUnmount = () => {}',
+      '  componentDidCatch = () => {}',
+      '  myInstanceMethod = () => {}',
+      '  render = () => (<div>Hello</div>)',
+      '}'
+    ].join('\n'),
+    parser: 'babel-eslint'
```

Comment:
Please add a test case that parses with the default parser and also covers all the new lifecycle methods.

Reply:
As far as I know I don't think I can test `getDerivedStateFromProps` with `React.createClass`. Is there a way I could specify a static method within the class body? I can write one for the other new methods though.

Reply:
`class Foo {} Foo.getDerivedStateFromProps = …` should work fine outside the class body

Reply:
Ya I meant that writing it that way wouldn't test anything with this rule since it wouldn't be sorted within the component body. I checked earlier and if I remember correctly it doesn't get caught by the rule in that case (as one would expect) or in the case where it is placed within `statics` (which I am not sure is for declaring static methods anyway [since I don't have much experience with the `React.createClass` style of React]). I can just leave it out though, or preferably add it after the component definition like you mentioned to make sure that doesn't get caught as an error in future versions.

Reply:
ah true. you could do `static getDerivedStateFromProps() { }` since it's a method tho?

Reply:
Oh, for some reason I didn't realize the default parser based tests had support for basic ES6 classes.